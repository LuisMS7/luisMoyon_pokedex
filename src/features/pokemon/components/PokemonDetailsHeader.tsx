import React from "react";
import { Pressable, StyleSheet, Text, View } from "react-native";
import { Pokemon } from "@pokemon/types/pokemon.type";
import { useNavigation } from "@react-navigation/native";
import { PokemonNavigationType } from "@pokemon/types/pokemon-navigator.type";
import { BOLD_WEIGHT, TEXT_BIG_SIZE, TEXT_SUPER_SIZE } from "@lib/constants";
import { titlecase } from "@utils/string-formatter";
import { ArrowLeftIcon } from "@pokemon/assets/LeftArrowIcon";
import { perfectSize } from "@utils/perfect-size";
import BigPokeballIcon from "@pokemon/assets/BigPokeballIcon";

type PokemonDetailsHeaderProps = { pokemon: Pokemon };

const PokemonDetailsHeader = ({ pokemon }: PokemonDetailsHeaderProps) => {
	const navigation =
		useNavigation<PokemonNavigationType<"PokemonDetailsScreen">>();

	const goBack = () => navigation.goBack();

	return (
		<View style={styles.container}>
			<View style={styles.pokeballIcon}>
				<BigPokeballIcon />
			</View>
			<Pressable onPress={goBack} testID="back-button">
				<ArrowLeftIcon />
			</Pressable>
			<Text style={styles.nameText}>{titlecase(pokemon.name)}</Text>
			<Text style={styles.idText}>
				#{pokemon.id.toString().padStart(4, "0")}
			</Text>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		flexDirection: "row",
		width: "100%",
		alignItems: "center",
		paddingHorizontal: 20,
	},
	pokeballIcon: {
		position: "absolute",
		top: 0,
		right: 0,
	},
	nameText: {
		fontSize: TEXT_SUPER_SIZE,
		fontWeight: BOLD_WEIGHT,
		color: "white",
		flex: 1,
		paddingLeft: perfectSize(7),
	},
	idText: {
		fontSize: TEXT_BIG_SIZE,
		fontWeight: BOLD_WEIGHT,
		color: "white",
	},
});

export default PokemonDetailsHeader;
