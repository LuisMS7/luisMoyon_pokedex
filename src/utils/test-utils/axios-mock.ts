import client from "@lib/axios-instance";
import MockAdapter from "axios-mock-adapter";

export const mockClient = new MockAdapter(client);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const mockGetOnSuccess = (endpoint: string, mockResponse: any) => {
	mockClient.onGet(endpoint).reply(200, mockResponse);
};

export const mockGetOnFailure = (endpoint: string, statusCode: number) => {
	mockClient.onGet(endpoint).reply(statusCode);
};
