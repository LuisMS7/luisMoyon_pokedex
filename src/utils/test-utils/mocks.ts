import { Pokemon } from "@pokemon/types/pokemon.type";

export const mockPokemon: Pokemon = {
	id: 1,
	name: "mockPokemonName",
	// eslint-disable-next-line max-len
	img: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
	types: [{ type: { name: "normal", url: "" }, slot: 1 }],
	weight: 10,
	sprites: {
		back_default:
			// eslint-disable-next-line max-len
			"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
		back_female: "",
		back_shiny: "",
		back_shiny_female: "",
		front_default: "",
		front_female: "",
		front_shiny: "",
		front_shiny_female: "",
	},
	moves: [{ move: { name: "normal", url: "" }, version_group_details: [] }],
};
