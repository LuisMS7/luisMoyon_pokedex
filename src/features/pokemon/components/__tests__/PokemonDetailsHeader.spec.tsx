import React from "react";
import {
	fireEvent,
	render,
	screen,
	waitFor,
} from "@testing-library/react-native";
import PokemonDetailsHeader from "@pokemon/components/PokemonDetailsHeader";
import { mockPokemon } from "@utils/test-utils/mocks";

const mockGoBack = jest.fn();

jest.mock("@react-navigation/native", () => ({
	useNavigation: () => ({
		goBack: mockGoBack,
	}),
}));

describe("PokemonDetailsHeader", () => {
	it("should render correctly", () => {
		render(<PokemonDetailsHeader pokemon={mockPokemon} />);
		expect(screen.toJSON()).toMatchSnapshot();
	});

	it("should go back when press left arrow icon", async () => {
		const { getByTestId } = render(
			<PokemonDetailsHeader pokemon={mockPokemon} />,
		);
		const leftArrowIcon = getByTestId("back-button");

		fireEvent(leftArrowIcon, "press");

		waitFor(() => expect(mockGoBack).toBeCalledTimes(1));
	});
});
