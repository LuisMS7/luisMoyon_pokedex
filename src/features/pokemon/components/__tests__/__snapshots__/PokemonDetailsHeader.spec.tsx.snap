// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetailsHeader should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="208"
      bbWidth="206"
      fill="none"
      focusable={false}
      height="208"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 208,
            "width": 206,
          },
        ]
      }
      vbHeight={208}
      vbWidth={206}
      width="206"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGGroup
          opacity={0.1}
        >
          <RNSVGPath
            d="M127.762 104C127.762 117.676 116.676 128.762 103 128.762C89.3244 128.762 78.2381 117.676 78.2381 104C78.2381 90.3244 89.3244 79.2381 103 79.2381C116.676 79.2381 127.762 90.3244 127.762 104Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
          <RNSVGPath
            clipRule={0}
            d="M103 208C155.393 208 198.738 169.257 205.947 118.857H145.035C138.917 136.169 122.407 148.571 103 148.571C83.5933 148.571 67.0835 136.169 60.9648 118.857H0.0532056C7.26233 169.257 50.6067 208 103 208ZM60.9648 89.1429H0.0532056C7.26233 38.7431 50.6067 0 103 0C155.393 0 198.738 38.7431 205.947 89.1429H145.035C138.917 71.8314 122.407 59.4286 103 59.4286C83.5933 59.4286 67.0835 71.8314 60.9648 89.1429ZM127.762 104C127.762 117.676 116.676 128.762 103 128.762C89.3244 128.762 78.2381 117.676 78.2381 104C78.2381 90.3244 89.3244 79.2381 103 79.2381C116.676 79.2381 127.762 90.3244 127.762 104Z"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="back-button"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="32"
      bbWidth="24"
      fill="none"
      focusable={false}
      height="32"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 32,
            "width": 24,
          },
        ]
      }
      vbHeight={32}
      vbWidth={24}
      width="24"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M12.2305 24.5117L13.0039 23.7383C13.1869 23.5553 13.1869 23.2585 13.0039 23.0754L6.94408 17.0157H20.2812C20.5401 17.0157 20.75 16.8058 20.75 16.5469V15.4532C20.75 15.1943 20.5401 14.9844 20.2812 14.9844H6.94408L13.0039 8.92465C13.1869 8.7416 13.1869 8.4448 13.0039 8.26172L12.2305 7.48836C12.0474 7.30531 11.7506 7.30531 11.5675 7.48836L3.38729 15.6686C3.20424 15.8516 3.20424 16.1484 3.38729 16.3315L11.5675 24.5118C11.7506 24.6948 12.0474 24.6948 12.2305 24.5117Z"
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    style={
      {
        "color": "white",
        "flex": 1,
        "fontSize": 47.71186567428981,
        "fontWeight": "800",
        "paddingLeft": 10.436970616250896,
      }
    }
  >
    Mockpokemonname
  </Text>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 23.855932837144906,
        "fontWeight": "800",
      }
    }
  >
    #
    0001
  </Text>
</View>
`;
